snapcraft/internal/lxd.py

```diff
@@ -219,7 +219,7 @@ def _finish(self):
 
     def execute(self, step='snap', args=None):
         super().execute(step, args)
-        if step == 'clean' and not args:
+        if step == 'clean' and 'pull' in args:
```

Comment:
Don't we have to check here if there are no parts in `args`? Maybe, this clean command is to clean only one part, and leave the rest.